lm_eval/models/sg_llm.py

```diff
@@ -79,15 +102,15 @@ def __init__(
             # These are the additional args to pass to the noisy model
             **noisy_model_args,
         )
-        if load_in_8bit:
+        if quantization_type != QuantizationType.default:
             noise_layer_prefix = "noise_layer."
             noisy_model.noise_layer.load_state_dict(
                 {k.replace(noise_layer_prefix, "", 1): v for k, v in state_dict["state_dict"].items() if k.startswith(noise_layer_prefix)}
             )
 
         else:
             noisy_model.load_state_dict(state_dict["state_dict"])
-        if bfloat_16 and not load_in_8bit:
+        if bfloat_16 and quantization_type == QuantizationType.default:
             noisy_model.base_model.to(dtype=torch.bfloat16)
```

Comment:
These 2 conditions both check `quantization_type == QuantizationType.default`, so you should probably combine them:

```suggestion
        if quantization_type == QuantizationType.default:
             noisy_model.load_state_dict(state_dict["state_dict"])
             if bfloat_16:
                 noisy_model.base_model.to(dtype=torch.bfloat16)
        else:
            noise_layer_prefix = "noise_layer."
            noisy_model.noise_layer.load_state_dict(
                {k.replace(noise_layer_prefix, "", 1): v for k, v in state_dict["state_dict"].items() if k.startswith(noise_layer_prefix)}
            )
```

Why do you only load the noise layer if `quantization_type != QuantizationType.default`? What about the base_model?

Reply:
I'll retain this for backward compatibility (don't want to affect folks' workflows and force a migration)

Reply:
What I suggested is logically equivalent but without explicitly checking both `quantization_type == QuantizationType.default` and `quantization_type != QuantizationType.default`

Reply:
Ok, checked and changed.